### ISSUE TITLE
Wrong position for added ticks x-axis after zoom

### ISSUE THREAD

**Issue by kzoon** [highcharts/highcharts#7463]
#### Expected behaviour
When using the tickposition option, the x-axis sometimes adds an additional tick. This tick should be in the right position 

#### Actual behaviour
1 The added tick can be outside the zoomed range
2 The added tick sometimes is not on a category

#### Live demo with steps to reproduce
1 The added tick can be outside the zoomed range: http://jsfiddle.net/kzoon/suyokcuL/1/:
zoom between august 2011 and April 2012 gives a tick on January 2012, but zoom between Jan 2011 and august 2011 gives no tick at all.

2 The added tick sometimes is not on a category: 
http://jsfiddle.net/kzoon/yhagvbg6/1/

#### Affected browser(s)
All

#### My analysis

Caused by axis trimTicks method:
Problem 1 is caused because the added tick is calculated by calculation the mean of  the first and the last item in tickPositions, disregarding the fact that it might be outside the current zoom range. Solution: 
Use the actual max and min of the axis
Problem 2 is caused because the calculated mean is not rounded

This code is my workaround, but should probably be polished:
   ```
       if (tickPositions.length === 0 && defined(roundedMin)) {
                if (this.categories) {
                    tickPositions.push(Math.round((this.min + this.max) / 2));
                } else {
                    tickPositions.push((roundedMax + roundedMin) / 2);
                }
            }
```

**Comment by KacperMadej** [comment]
Hi,

Thank you for reporting about the problem.
In general use `tickPositioner` if outcome of `tickPositions` option is not working as you need.

**Issue summed up:**
* number ticks for category type axis are not acceptable - check: https://github.com/highcharts/highcharts/blob/c904e1f931ed5541f317e3be3443793360723a5f/js/parts/Axis.js#L3254-L3257
* possibly disable the fix for #3195 if `tickPositions` are given from user options

**Comment by TorsteinHonsi** [comment]
In my opinion, there should be no extra tick inserted at all in this case, because the tick positions were explicitly set.
